lib/Symfony/SearchBundle/DependencyInjection/Compiler/ElasticaBundlePass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the RollerworksSearch package.
+ *
+ * (c) Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Rollerworks\Bundle\SearchBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * If FOS Elastica Bundle is detected, use its client as our client.
+ *
+ * Class ElasticaBundlePass.
+ */
+class ElasticaBundlePass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('fos_elastica.client.default')) {
```

Comment:
@sstok can you help me out here? For some reason, this will always return false (as if FOS Elastica is not loaded, even though it's registered) and I'm getting

```
1) Rollerworks\Bundle\SearchBundle\Tests\Functional\ApiPlatformTest::testEmptySearchCodeIsValid
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "rollerworks_search.elasticsearch.client". Did you mean this: "rollerworks_search.elasticsearch.factory"?
```

Reply:
Maybe try `has()` if it's an alias it might not be detected 🤔

Reply:
No, nothing related to Elastica is found for some reason. I'm trying to debug this and hitting a wall, can't figure it out. :(

Reply:
Bundle is definetly registered as I'm seeing compiler passes from it further down the list if I dump the compiler passes from the entire app in the compiler. I've also tried all the `PassConfig` types, it should work without any of that but it doesn't.